Index: exo.kernel.commons/src/test/java/org/exoplatform/commons/utils/TestLogMessageFormatter.java
===================================================================
--- exo.kernel.commons/src/test/java/org/exoplatform/commons/utils/TestLogMessageFormatter.java	(revision 5990)
+++ exo.kernel.commons/src/test/java/org/exoplatform/commons/utils/TestLogMessageFormatter.java	(working copy)
@@ -57,4 +57,17 @@
       assertEquals(t, LogMessageFormatter.getThrowable(new Object(), t));
       assertEquals(t, LogMessageFormatter.getThrowable("Just testing{}", new Object(), t));
    }
+
+   public void testMessageIsNull() throws Exception
+   {
+      LogMessageFormatter.getMessage("Hello, World {} ", new A());
+   }
+
+   class A
+   {
+      public String toString()
+      {
+         return null;
+      }
+   }
 }
Index: exo.kernel.commons/src/main/java/org/exoplatform/services/log/impl/LogMessageFormatter.java
===================================================================
--- exo.kernel.commons/src/main/java/org/exoplatform/services/log/impl/LogMessageFormatter.java	(revision 5990)
+++ exo.kernel.commons/src/main/java/org/exoplatform/services/log/impl/LogMessageFormatter.java	(working copy)
@@ -41,13 +41,11 @@
          {
             if (i != argsArray.length - 1 || !(argsArray[i] instanceof Throwable))
             {
-               str = REPLACE_PATTERN.matcher(str).replaceFirst(String.valueOf(argsArray[i]));
+               str = REPLACE_PATTERN.matcher(str).replaceFirst(String.valueOf(argsArray[i].toString()));
             }
          }
       }
       return str;
-
-
    }
 
    public static Throwable getThrowable(Object... argsArray)
