Index: exo.kernel.container/src/main/java/org/exoplatform/container/RootContainer.java
===================================================================
--- exo.kernel.container/src/main/java/org/exoplatform/container/RootContainer.java	(revision 5937)
+++ exo.kernel.container/src/main/java/org/exoplatform/container/RootContainer.java	(working copy)
@@ -30,7 +30,7 @@
 import org.exoplatform.container.xml.Configuration;
 import org.exoplatform.management.annotations.Managed;
 import org.exoplatform.management.annotations.ManagedDescription;
-import org.exoplatform.management.jmx.annotations.NamingContext;
+import org.exoplatform.management.jmx.annotations.NameTemplate;
 import org.exoplatform.management.jmx.annotations.Property;
 import org.exoplatform.management.rest.annotations.RESTEndpoint;
 import org.exoplatform.services.log.ExoLogger;
@@ -56,7 +56,7 @@
  * tuan08@users.sourceforge.net Date: Jul 21, 2004 Time: 12:15:28 AM
  */
 @Managed
-@NamingContext(@Property(key = "container", value = "root"))
+@NameTemplate(@Property(key = "container", value = "root"))
 @RESTEndpoint(path = "rcontainer")
 public class RootContainer extends ExoContainer
 {
Index: exo.kernel.container/src/main/java/org/exoplatform/container/StandaloneContainer.java
===================================================================
--- exo.kernel.container/src/main/java/org/exoplatform/container/StandaloneContainer.java	(revision 5937)
+++ exo.kernel.container/src/main/java/org/exoplatform/container/StandaloneContainer.java	(working copy)
@@ -27,7 +27,7 @@
 import org.exoplatform.container.xml.Configuration;
 import org.exoplatform.management.annotations.Managed;
 import org.exoplatform.management.annotations.ManagedDescription;
-import org.exoplatform.management.jmx.annotations.NamingContext;
+import org.exoplatform.management.jmx.annotations.NameTemplate;
 import org.exoplatform.management.jmx.annotations.Property;
 import org.exoplatform.management.rest.annotations.RESTEndpoint;
 
@@ -51,7 +51,7 @@
  *          (from where JVM is started) for standalone. See
  */
 @Managed
-@NamingContext(@Property(key = "container", value = "standalone"))
+@NameTemplate(@Property(key = "container", value = "standalone"))
 @RESTEndpoint(path = "scontainer")
 public class StandaloneContainer extends ExoContainer implements SessionManagerContainer
 {
